src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,31 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
-            if (argNode->gtRegNum != argReg)
+#ifdef _TARGET_ARM_
+            if (curArgTabEntry->isSplit)
             {
-                inst_RV_RV(ins_Move_Extend(argNode->TypeGet(), true), argReg, argNode->gtRegNum);
+                assert(curArgTabEntry->numRegs >= 1);
+                for (unsigned idx = 0; idx < curArgTabEntry->numRegs; idx++)
+                {
+                    regNumber regDst = (regNumber)((unsigned)curArgTabEntry->regNum + idx);
```

Comment:
nit: instead of `regDst`, I would use `argReg`, since that is used elsewhere in the function to convey the same concept.